File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/LouseDefensive_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class LouseDefensive_Model extends MonsterInfo{
    public static Id = "LouseDefensive"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(12,18);
        else
            return Mathf.RandomRange(11,17);
    }

    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/louseGreen/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/louseGreen/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/louseGreen/skeleton.png"));
    }

    get Name(): string {
        return "防御型虱虫";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/LouseDefensive_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class LouseDefensive_Model : MonsterInfo
    {
        public static string Id = "LouseDefensive";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(12,18) : Random.Range(11,17);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/louseGreen/skeleton.atlas",
                    "Monsters/theBottom/louseGreen/skeleton.json", "Monsters/theBottom/louseGreen/skeleton.png");
            }
        }

        public override string Name => "防御型虱虫";

        public LouseDefensive_Model(Vector2 pos) : base(pos)
        {
        }
    }
}